### PR TITLE
Fybrik - adding ArgoCD application manifests for Vault and Arrow Flight Module

### PR DESCRIPTION
**Pull request by eletonia**
This PR adds the ArgoCD application manifests to deploy Vault and Arrow Flight Module on the Smaug cluster

@ronenkat @HumairAK

### CODE REVIEW

File: argocd/overlays/moc-infra/applications/envs/moc/smaug/fybrik/arrow-flight-module.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+project: fybrik
```
Comment (HumairAK) [inline argocd/overlays/moc-infra/applications/envs/moc/smaug/fybrik/arrow-flight-module.yaml:1]: you seem to be missing the metadata, apiversion, and spec fields (same with vault)

Comment (eletonia) [reply argocd/overlays/moc-infra/applications/envs/moc/smaug/fybrik/arrow-flight-module.yaml]: @HumairAK I just added the required fields

**Review by HumairAK — APPROVED** [review]
/lgtm
/approve

**Comment by HumairAK** [comment]
lgtm but can you squash your commits?

**Comment by eletonia** [comment]
> lgtm but can you squash your commits, and address the linting errors?

@HumairAK Done

**Pull request merged**